lms/djangoapps/bulk_email/models.py

```diff
@@ -107,10 +110,45 @@ def create(cls, course_id, sender, to_option, subject, html_message, text_messag
             template_name=template_name,
             from_addr=from_addr,
         )
+
+        if not course_email.from_addr:
+            course_email.calculate_default_from_addr()
+
         course_email.save_now()
 
         return course_email
 
+    def calculate_default_from_addr(self):
+        """
+        Calculates an email address to be used as the 'from-address' for sent emails.
+
+        Makes a unique from name and address for each course, e.g.
+
+            "COURSE_TITLE" Course Staff <coursenum-no-reply@courseupdates.edx.org>
+
+        """
+        # Fetch the course object.
+        course = get_course(self.course_id)
+
+        if course is None:
+            msg = u"CourseEmail %s: course not found: %s"
+            log.error(msg, self.subject, self.course_id)
+            raise ValueError(msg % (self.subject, self.course_id))
+
+        course_title = course.display_name
+        course_title_no_quotes = re.sub(r'"', '', course_title)
+
+        # For the email address, get the course.  Then make sure that it can be used
+        # in an email address, by substituting a '_' anywhere a non-(ascii, period, or dash)
+        # character appears.
+        # Translators: This is the email subject when the course staff emails the students
+        from_addr = _('"{course_title}" Course Staff <{course_id}-{from_email}>').format(
```

Comment:
@OmarIthawi I'm wary about this change. I would prefer that the default is not the instructor's language but the platform default language. We had a similar discussion on #6082:

"I feel that the right thing to do is actually to default to the _platform_ language, not the instructor's language. For example there is a France-based installation and Professor Yang is teaching a course in French on this France-based installation. But he is from China so he sets his own platform language to be Chinese. I think it is more appropriate to default not to his language but to the platform language."